netkan/.gitconfig

```diff
@@ -1,3 +1,10 @@
 [user]
 	email = netkan-bot@ksp-ckan.space
 	name = NetKAN inflator Robot
+
+[gc]
+	auto = 2700
```

Comment:
The "expire" settings look like exactly what we need. :100: :rocket: 

This "pack" thing I'm less sure about. It sounds like a way to _keep_ garbage and shuffle it around rather than eliminating it. (The default tuning seems much more suited for Linus maintaining the kernel than an automated service that's never going to need to dig up old loose commits.)

Should we try to turn that off and **only** expire garbage? It's concerning that the documentation says "Setting this to 0 disables not only automatic packing based on the number of loose objects, but any other heuristic git gc --auto will otherwise use to determine if there’s work to do", implying that this has to be enabled to run auto gc at all, but maybe the other settings can be tuned to avoid packing.

Reply:
From my reading, this turns off auto gc entirely:
>  but any other heuristic git gc --auto will otherwise use to determine if there’s work to do
I think from our use case, pruning will always be required. I don't know automatic packing is a big deal, it's pretty quick and a setting of 2700 will mean that the whole lot should run every few hours (it was at around 2800 after 3 hours).

What happens under the hood is that copies get added to the disk and git has to follow each of these when doing operations (likely why when there are lot of branches, it has a lot more work to do following all these to check for references). Packing brings them into contiguous "packs". I think that's something we definitely want and it was already doing that reasonably frequently.

https://git-scm.com/book/en/v2/Git-Internals-Packfiles

Reply:
> From my reading, this turns off auto gc entirely:

Agreed, I am definitely **not** suggesting setting `gc.auto` to 0. I was more wondering if there might be a configuration of `autoPackLimit` and `bigPackThreshold` and `packRefs` that would ensure that garbage is expired/pruned instead of "packed". But admittedly I'm guessing a lot about what "pack" means, since based on this documentation it might or might not have anything to do with garbage (though then why it's related to `git gc` becomes a big question mark).